drivers/pipes/pipe_common.c

```diff
@@ -171,56 +173,109 @@ int pipecommon_open(FAR struct file *filep)
     {
       dev->d_nwriters++;
 
-      /* If this is the first writer, then the read semaphore indicates the
-       * number of readers waiting for the first writer.  Wake them all up.
+      /* If this is the first writer, then the n-writers semaphore
+       * indicates the number of readers waiting for the first writer.
+       * Wake them all up!
        */
 
       if (dev->d_nwriters == 1)
         {
-          while (nxsem_get_value(&dev->d_rdsem, &sval) == 0 && sval <= 0)
+          while (nxsem_get_value(&dev->d_nwrsem, &sval) == OK && sval <= 0)
             {
-              nxsem_post(&dev->d_rdsem);
+              nxsem_post(&dev->d_nwrsem);
             }
         }
     }
 
+  while ((filep->f_oflags & O_NONBLOCK) == 0 &&     /* Blocking */
+         (filep->f_oflags & O_RDWR) == O_WRONLY &&  /* Write-only */
```

Comment:
move to the if block at line 172 and remove O_WRONLY check

Reply:
I'm afraid that the logic wouldn't be the same if we move this `while` block into `if` block at 172 + removing the `O_WRONLY` check: 

`if ((filep->f_oflags & O_WROK) != 0)` only checks if the file is open for writing (it could be open for writing and reading and this check would still be valid). But the logic for blocking open, according to IEEE Std 1003.1-2017, is valid only when is being opened for writing/reading-only (`(filep->f_oflags & O_RDWR) == O_WRONLY` assures that). An intermediary solution would be moving but letting the `O_WRONLY` check.

Am I missing something?

Reply:
Yes, you are right. But I think we can move the while loop into if block.

Reply:
I'd prefer to make it on a separate patch. The `O_RDONLY` check is already existing and resides outside of the corresponding `if` block. I'll just add the `O_WRONLY` while similarly. Another patch can move them and create specific semaphores (https://github.com/apache/nuttx/pull/8985#discussion_r1164328463). Do you agree?